File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.cpp
#include <bits/stdc++.h>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    set<string> s(phone_book.begin(), phone_book.end());
    
    for(auto iter = s.begin(); iter != s.end(); iter++) {
        string lhs = *iter++;
        if(iter == s.end()) break;
        string rhs = *iter--;
        
        if(lhs.length() > rhs.length()) continue;
        
        string rhs_sub = rhs.substr(0, lhs.length());
        if(lhs.compare(rhs_sub) == 0) {
            answer = false;
            break;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    
    phone_book.sort()
    
    length = len(phone_book)
    
    for i in range(0, length - 1):
        if phone_book[i + 1].startswith(phone_book[i]):
            answer=False
            break
    
    return answer